text/trace/0247-peer-service-propagation.md

```diff
@@ -0,0 +1,157 @@
+# Automatic Peer Service Name propagation
+
+Automatic propagation of `peer.service` through `TraceState`.
+
+## Motivation
+
+Knowing the service name on the other side of a remote call is valuable
+troubleshooting information. The semantic conventions represent this via
+`peer.service`, which needs to be manually populated. In a deployment scenario,
+when a new service is added, all the existing services interacting with it
+need to update `peer.service`, which is error-prone and may
+become unreliable, making it eventually obsolete.
+
+This information can be effectively derived in the backend using the
+`Resource` of the parent `Span`, but is otherwise not available
+at Collector processing time, where it could be used for sampling
+and tranformation purposes.
+
+As metrics and logs do not have defined a parent-child relationship, using
+`peer.service` could help gaining insight into the remote service as well.
+
+Defining (optional) automated population of `peer.service` will greatly help
+adoption of this attribute by users and vendors explictly interested in this
+scenario.
+
+## Explanation
+
+SDKs will define an optional feature, disabled by default,
+to read the `service.name` attribute of the related `Resource` and set it
+in the spans' `TraceState` as described in
+[trace state handling](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-handling.md)
+specifically using the `sn` subkey (denoting **service name**):
+
+```
+ot=sn:MyService
```

Comment:
`sn` feels fishy to me because of uncertain lifecycle scope. From the moment the request from sn=A is received by service B, when does `state.sn=A` change in service B's request execution? I assume only when B makes downstream calls. So during all the B execution `sn` confusingly refers to A.

Suggestion: `c` for `caller` or `us` for `upstream service`

Reply:
Applied, using `us` for now.